File: backend/BowlerAPI/Data/Team.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace BowlerAPI.Data;

public partial class Team
{
    [Key]
    [Required]
    public int TeamId { get; set; }
    [Required]
    public string TeamName { get; set; } = null!;

    public int? CaptainId { get; set; }
    [JsonIgnore]
    public virtual ICollection<Bowler> Bowlers { get; set; } = new List<Bowler>();

    public virtual ICollection<TourneyMatch> TourneyMatchEvenLaneTeams { get; set; } = new List<TourneyMatch>();

    public virtual ICollection<TourneyMatch> TourneyMatchOddLaneTeams { get; set; } = new List<TourneyMatch>();
}

File: frontend1/src/types/Team.ts
import { Bowler } from './Bowler';
import { TourneyMatch } from './TourneyMatch';

export type Team = {
  teanId: number;
  teamName: string;
  captainId: number;
  bowlers: Bowler[];
  tourneyMatchEvenLaneTeams: TourneyMatch[];
  tourneyMatchOddLaneTeams: TourneyMatch[];
};
